frontend/components/SurplusAuction.vue

```diff
@@ -0,0 +1,247 @@
+import FormatCurrency from '~/components/utils/FormatCurrency.vue';
+import AuctionRestartPanel from '~/components/panels/AuctionRestartPanel.vue';
+import LoadingIcon from '~/assets/icons/loading.svg';
+export default Vue.extend({
+    name: 'SurplusAuction',
+    components: {
+        AuctionRestartPanel,
+        PriceDropAnimation,
+        FormatCurrency,
+        TextBlock,
+        TimeTill,
+        Button,
+        FormatMarketValue,
+        Alert,
+        Tooltip,
+        LoadingIcon,
+    },
+    props: {
+        auction: {
+            type: Object as Vue.PropType<SurplusAuction>,
+            default: null,
+        },
+        auctionId: {
+            type: String,
+            required: true,
+        },
+        error: {
+            type: String,
+            default: '',
+        },
+        isExplanationsShown: {
+            type: Boolean,
+            default: true,
+        },
+        walletAddress: {
+            type: String,
+            default: null,
+        },
+    },
+    data() {
+        return {
+            isTableExpanded: false,
+        };
+    },
+    computed: {
+        auctionError(): { error: string; showBanner: boolean } | null {
+            if (this.error) {
+                return {
+                    error: this.error,
+                    showBanner: true,
+                };
+            } else if (!this.auction) {
+                return {
+                    error: 'This auction was not found',
+                    showBanner: true,
+                };
+            } else if (this.auction.state === 'ready-for-collection' || this.auction.state === 'collected') {
+                return {
+                    error: 'This auction is finished',
+                    showBanner: true,
+                };
+            } else if (this.auction.state === 'requires-restart') {
+                return {
+                    error: 'This auction is inactive and must be restarted',
+                    showBanner: false,
+                };
+            }
+            return null;
+        },
+        isActive(): boolean {
+            return this.auction.state === 'just-started' || this.auction.state === 'have-bids';
+        },
+        requiresRestart(): boolean {
+            return this.auction.state === 'requires-restart';
+        },
+        unitPrice(): BigNumber {
+            return this.auction.bidAmountMKR.dividedBy(this.auction.receiveAmountDAI);
+        },
+    },
+    watch: {
+        isExplanationsShown: {
+            immediate: true,
+            handler(newIsExplanationsShown) {
+                if (!newIsExplanationsShown) {
+                    this.isTableExpanded = true;
+                }
+            },
+        },
+        areAuctionsFetching(areAuctionsFetching) {
+            if (!areAuctionsFetching && !this.auction && !this.error) {
+                this.$emit('fetchTakeEventsFromAuction', this.auctionId);
```

Comment:
Let us remove this line for now. For reference, this line is called when the auction could not be found. In the case of swap auctions, we fetch the take events (all the transactions used to take up the auction), however I don't know what we want to fetch for expired surplus auctions.